lib/sacastats_web.ex

```diff
@@ -78,6 +78,36 @@ defmodule SacaStatsWeb do
     end
   end
 
+  def generate_page_title(endpoint) do
+    site_name = "Saca Stats"
+    separator = " - "
+    postfix = separator <> site_name
+    endpoint_arr = String.split(endpoint, "/", trim: true)
+
+    case length(endpoint_arr) do
+      0 -> site_name
+      _ -> generate_subpage_title(endpoint_arr, separator) <> postfix
+    end
+  end
+
+  def generate_subpage_title([first | rest], separator)
+      when first == "character" and length([first | rest]) > 1,
+      do: generate_character_title([first | rest], separator)
```

Comment:
Nice use of pattern matching/guards! If you want you could take this further and replace `first` with `"character"` to reduce the first conditional in the guard. you could also match `[first | rest] = endpoint_arr` so you can use `endpoint_arr` in other places instead of rewriting `[first | rest]`.
```suggestion
  def generate_subpage_title(["character" | _rest] = endpoint_arr, separator)
      when length(endpoint_arr) > 1,
      do: generate_character_title(endpoint_arr, separator)
```

Reply:
Ah okay, that makes sense.